jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/service/MigrateServiceTest.java

```diff
@@ -17,90 +17,93 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.function.Consumer;
 
 import org.eclipse.jkube.kit.common.KitLogger;
 
 import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.IOUtils;
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
-import org.junit.runner.RunWith;
-import org.mockito.Mock;
-import org.mockito.junit.MockitoJUnitRunner;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
-@RunWith(MockitoJUnitRunner.class)
-public class MigrateServiceTest {
-  @Mock
-  KitLogger logger;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
 
-  @Rule
-  public TemporaryFolder folder = new TemporaryFolder();
+class MigrateServiceTest {
+  private KitLogger logger;
+
+  @TempDir
+  Path folder;
```

Comment:
I mentioned this in this example because most test cases are just using the file.

> But I don't know how to do these type of things 

For a one level nested file you can simply do `new File(folder, "path-to-file-or-dir.ext");`

For an n-leveled nested file you can do `folder.toPath().resolve("directory").resolve("file2).xxx`, which is more or less the opposite of what you're doing now so it might or might not be suitable.

The point here is that (IMHO) we should go for Path or File trying to minimize the number of path.toFile or file.toPath calls.


> with File type or create temporary directories with File. Also it makes migration easier especially for TemporaryFolder

This is confusing. I'm not sure if you understood what I mean:
```java
@TempDir
Path directory;
@TempDir
File directoryF;
```
both options work.